src/main/java/huyang/Huyang.java

```diff
@@ -0,0 +1,157 @@
+                ArrayList<Task> foundTasks = tasks.findTasks(keyword);
+                return ui.getFoundTasksMessage(foundTasks);
+            case BYE:
+                PauseTransition delay = new PauseTransition(Duration.seconds(1));
+                delay.setOnFinished(event -> Platform.exit());
+                delay.play();
+                return ui.getFarewellMessage();
+            case UNKNOWN:
+            default:
+                return ui.getUnknownCommandMessage();
+            }
+        } catch (TaskException | IOException e) {
+            return ui.getErrorMessage(e.getMessage());
+        }
+    }
+
+    /**
+     * Starts the Huyang chatbot.
+     * Displays a greeting, processes user commands, and saves tasks to a file.
+     * Supports other features such as list, mark, unmark and delete.
+     */
+    public void runCli() {
+        ui.print(ui.getGreetingMessage());
+        Scanner scanner = new Scanner(System.in);
+        boolean isExit = false;
+
+        while (!isExit) {
+            String input = scanner.nextLine();
+            Parser.CommandType command = parser.parseCommand(input);
+
+            try {
+                switch (command) {
+                case LIST:
+                    ui.print(ui.getTasksMessage(tasks.getTasks()));
+                    break;
+                case MARK:
+                    Task markedTask = tasks.markOrUnmarkTask(input, true);
+                    ui.print(ui.getMarkOrUnmarkMessage(markedTask, true));
+                    storage.saveTasks(tasks.getTasks());
+                    break;
+                case UNMARK:
+                    Task unmarkedTask = tasks.markOrUnmarkTask(input, false);
+                    ui.print(ui.getMarkOrUnmarkMessage(unmarkedTask, false));
+                    storage.saveTasks(tasks.getTasks());
+                    break;
+                case TODO:
+                case DEADLINE:
+                case EVENT:
+                    Task addedTask = tasks.addTask(input, command);
+                    ui.print(ui.getAddTaskMessage(addedTask, tasks.getSize()));
+                    storage.saveTasks(tasks.getTasks());
+                    break;
+                case DELETE:
+                    Task deletedTask = tasks.deleteTask(input);
+                    ui.print(ui.getDeleteTaskMessage(deletedTask, tasks.getSize()));
+                    storage.saveTasks(tasks.getTasks());
+                    break;
+                case FIND:
+                    String keyword = input.substring(5).trim().toLowerCase();
+                    ArrayList<Task> foundTasks = tasks.findTasks(keyword);
+                    ui.print(ui.getFoundTasksMessage(foundTasks));
+                    break;
+                case BYE:
+                    isExit = true;
+                    break;
+                case UNKNOWN:
+                default:
+                    ui.print(ui.getUnknownCommandMessage());
+                    break;
+                }
```

Comment:
I like the use of switch case here to make the code more readable and easier to debug 👍

Reply:
Thank you!